docs/graasp-insights-user-guide.md

```diff
@@ -0,0 +1,243 @@
+# Datasets
+
+## Menu navigation
+
+When opening Graasp-Insights, you are greeted with the `Datasets` tab.
+Use the top-left menu button to navigate to each distinctive tab.
+
+![Datasets tab with menu open](./figures/datasets-menu.png)
+
+## Importing a dataset
+
+Load a dataset into Graasp Insights by using the lower right `+` button.
+
+Choose a dataset to import by navigating to its location in your file system.
+You can optionally give it a name and description for its use in Graasp Insights.
+
+![Add dataset form](./figures/add-dataset.png)
+
+Importing a dataset will create a copy of the file and manipulating it through Graasp Insights will have no impact on the original file.
+Only `json`, `csv` and `xlsx` datasets are supported in the current state.
+
+## Dataset management
+
+Once loaded, general information about the dataset is displayed. It can additionally be viewed, exported and deleted using the quick actions.
+
+![Datasets tab with loaded dataset](./figures/loaded-dataset.png)
+
+Click on the `View dataset` button to inspect the content of the dataset.
+If the dataset follows the `Graasp` schema, additional information about the content is also displayed.
+
+Navigate through the data and edit any particular value with the `RAW` view.
+
+![Dataset screen](./figures/dataset-screen.png)
+
+If the dataset was loaded as a `csv` or `xlsx` file, or if the structure of the dataset supports it, you can switch to a `TABLE` view to display a tabular version of it.
+
+![Dataset screen (table view)](./figures/dataset-screen-table.png)
+
+# Algorithms
+
+The `Algorithms` tab contains every algorithm that can be used throughout Graasp Insights. It is also the entry point for accessing, editing and writing algorithms.
+
+![Algorithms tab](./figures/algorithms.png)
+
```

Comment:
Maybe:

_**Note**: algorithms in Graasp Insights are written in the Python programming language._